common/src/main/java/io/netty/util/internal/PlatformDependent.java

```diff
@@ -1347,8 +1347,18 @@ private static String normalizeOs(String value) {
         return "unknown";
     }
 
+    /**
+     * @deprecated As of release siren-4.1.27-4 this method will return 0L if {@link #javaVersion()} is 9+.
+     */
+    @Deprecated
     public static long getDirectMemoryCounter() {
-        return DIRECT_MEMORY_COUNTER.get();
+        if (PlatformDependent0.hasDirectBufferNoCleanerConstructor()) {
+            return DIRECT_MEMORY_COUNTER.get();
+        } else {
+            logger.warn("sun.misc.Unsafe or java.nio.DirectByteBuffer.<init>(long, int) not available, " +
+              "therefore there is no count direct memory allocation");
+            return 0L;
+        }
```

Comment:
I think we can remove this method, it was introduced by us at some point in time, see
https://github.com/netty/netty/blob/4.1/common/src/main/java/io/netty/util/internal/PlatformDependent.java#L1497